recursive-feature-elimination/script.whizzml

```diff
@@ -0,0 +1,248 @@
+                                                  "seed" (str "BigML Boruta" 
+                                                               run) }}))
+
+;; evaluate-model
+;;
+;; Creates and returns an evaluation for a model
+;;
+;; Inputs:
+;;   model-id: (model-id) Model to evaluate
+;;   test-ds-id: (dataset-id) Test Dataset id
+;; Output:
+;;   evaluation: (evaluation) Result of the model evaluation
+
+  (define (evaluate-model model test-ds-id)
+    (create-and-wait-evaluation {"dataset" test-ds-id
+                                "model" model}))
+
+
+  ;; recursive-feature-elimination
+  ;;
+  ;; Applies recursive feature elimination algorithm and returns a map with
+  ;; the selected fields names, selected fields ids, and information about
+  ;; each iteration
+  ;;
+  ;; Inputs:
+  ;;   dataset-id: (dataset-id) Dataset id (for training)
+  ;;   nfeatures: (number) Number of features that we want to obtain
+  ;;   objective-id: (string) Objective field id
+  ;;   test-dataset-id: (dataset-id) id of the dataset used to
+  ;;                     evaluate the model
+  ;;   evaluation-metric: (string) Metric  used in the evaluation
+  ;; Output: (map) Map with selected field names, selected field ids, and 
+  ;;         iterations evaluation info (only if test-dataset-id is present)
+  (define (recursive-feature-elimination dataset-id
+                                         nfeatures
+                                         objective-id
+                                         test-dataset-id
+                                         evaluation-metric)
+    (let (do-evaluation (not (empty? test-dataset-id))
+          obj-id (get-objective dataset-id objective-id)
+          fields (resource-fields dataset-id)
+          input-ids (default-inputs dataset-id obj-id))
+      (loop (selected input-ids
+             evaluations []) 
+          ;; Loop until we obtain the desired number of features
+          (if (<= (count selected) nfeatures)
+              (response selected 
+                        (feature-names fields selected) 
+                        evaluations 
+                        do-evaluation)
+              (let (_ (log-info "------- New iteration -------")
+                    _ (log-info "Creating random forest with " 
+                                (count selected) 
+                                " fields")
+                    ;; Create random forest and obtain its importances
+                    rf-id (random-forest dataset-id obj-id 0 selected)
+                    imps ((fetch rf-id) "importance")
+		    ;; Obtain fields with importance 0
+		    ;; (they  doesnt appear in "importance")
+		    no-imp-fields (filter (lambda (n)
+					    (not (member? n (keys imps))))
+					  selected)
+                    ;; Obtaining minimum field importance
+                    removed-field (min-key (lambda (x) (imps x)) (keys imps))
+		    removed-fields (cons removed-field no-imp-fields)
```

Comment:
ditto (i am assuming there are no TABs in the source, right?)

Reply:
In emacs, indentation looked good so, maybe yes, I may have used TAB in some place.
I have run untabify. (And I have written in big letters in my notebook TABS ARE EVIL)